src/logging/request-obfuscator.js

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * Module dependencies.
+ */
+
+import _ from 'lodash';
+import methods from '../methods';
+
+/**
+ * Lower case methods.
+ */
+
+const lowerCaseMethods = _.mapKeys(methods, (value, key) => key.toLowerCase());
+
+/**
+ * Obfuscate the response body.
+ */
+
+function obfuscateResponseBody(body, methodName) {
+  const method = _.get(lowerCaseMethods[methodName], 'obfuscate.response');
+
+  if (!method || _.isEmpty(body.result)) {
+    return body;
+  }
+
+  body.result = method(body.result);
+
+  return body;
+}
+
+/**
+ * Obfuscate the response.
+ */
+
+function obfuscateResponse(request, instance) {
+  if (_.get(request, 'response.headers[\'content-type\']') === 'application/octet-stream') {
+    request.response.body = '******';
+
+    return;
+  }
+
+  if (!instance.body) {
+    return;
+  }
+
+  const requestBody = JSON.parse(instance.body);
+
+  if (_.isArray(request.response.body)) {
+    if (_.get(request, 'response.headers[\'content-type\']') === 'application/octet-stream') {
+      request.response.body = _.map(request.response.body, request => _.set(request, 'result', '******'));
+
+      return;
+    }
+    const methodsById = _.mapKeys(requestBody, method => method.id);
+
+    request.response.body = _.map(request.response.body, request => obfuscateResponseBody(request, methodsById[request.id].method));
+
+    return;
+  }
+
+  request.response.body = obfuscateResponseBody(request.response.body, requestBody.method);
+}
+
+/**
+ * Obfuscate the request body.
+ */
+
+function obfuscateRequestBody(body) {
+  const method = _.get(lowerCaseMethods[body.method], 'obfuscate.request');
+
+  if (!method) {
+    return body;
+  }
+
+  body.params = method(body.params);
+
+  return body;
+}
+
+/**
+ * Obfuscate the request.
+ */
+
+function obfuscateRequest(request) {
+  request.body = JSON.parse(request.body);
+
+  if (_.isArray(request.body)) {
+    request.body = _.map(request.body, obfuscateRequestBody);
+  } else {
+    request.body = obfuscateRequestBody(request.body);
+  }
+
+  request.body = JSON.stringify(request.body);
+}
+
+/**
+ * Export `RequestObfuscator`.
+ */
+
+export function obfuscate(request, instance) {
+  if (_.has(request, 'headers.authorization')) {
+    request.headers.authorization = request.headers.authorization.replace(/(Basic )(.*)/, `$1******`);
+  }
+
+  if (_.isString(request.body)) {
```

Comment:
Perhaps move this test to the function? It already tests a lot of conditions there.